extensions/wikia/Insights/models/InsightsQuerypageModel.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * Class QueryPagesModel
+ *
+ * Model for pages which extends QueryPage
+ */
+abstract class InsightsQuerypageModel extends InsightsModel {
+	private $queryPageInstance,
+		$template = 'subpageList';
+
+	public
+		$offset = 0,
+		$limit = 100;
+
+	abstract function getDataProvider();
```

Comment:
All of them are part of the subpages based on QueryPage. I'd rather keep the InsightsModel minimal. We can think about refactoring InsightsModel to an Interface since it has only abstract methods.